documentation/docs/03-loading.md

```diff
@@ -0,0 +1,123 @@
+
+Our example blog page might contain a `load` function like this:
+
+```html
+<script context="module">
+	/**
+	 * @param {{
+	 *   page: {
+	 *     host: string;
+	 *     path: string;
+	 *     params: Record<string, string | string[]>;
+	 *     query: URLSearchParams;
+	 *   };
+	 *   fetch: (url: string, opts?: {...}) => Promise<Response>
+	 *   session: any;
+	 *   context: Record<string, any>;
+	 * }} options
+	 * @returns {{
+	 *   status?: number;
+	 *   error?: Error | string;
+	 *   redirect?: string;
+	 *   maxage?: number;
+	 *   props?: Record<string, any>;
+	 *   context?: Record<string, any>;
+	 * }}
+	 */
+	export async function load({ page, fetch, session, context }) {
+		const url = `/blog/${page.params.slug}.json`;
+		const res = await fetch(url);
+
+		if (res.ok) {
+			return {
+				props: {
+					article: await res.json()
+				}
+			};
+		}
+
+		return {
+			status: res.status,
+			error: new Error(`Could not load ${url}`)
+		};
+	}
+</script>
+```
+
+`load` is the SvelteKit equivalent of `getStaticProps` or `getServerSideProps` in Next.js or `asyncData` in Nuxt.js.
+
+Note the `context="module"` — this is necessary because `load` runs before the component is rendered.
+
+> `load` only applies to components that define pages, not the components that they import.
+
+### Input
+
+The `load` function receives an object containing four fields —  `page`, `fetch`, `session` and `context`.
+
+#### page
+
+`page` is a `{ host, path, params, query }` object where `host` is the URL's host, `path` is its pathname, `params` is derived from `path` and the route filename, and `query` is an instance of `URLSearchParams`.
+
+So if the example above was `src/routes/blog/[slug].svelte` and the URL was `/blog/some-post?foo=bar&baz`, the following would be true:
+
+- `page.path === '/blog/some-post'`
+- `page.params.slug === 'some-post'`
+- `page.query.get('foo') === 'bar'`
+- `page.query.has('baz')`
+
+#### fetch
+
+`fetch` is equivalent to the native `fetch` web API, and can make credentialled requests. It can be used across both client and server contexts.
+
+> When `fetch` runs on the server, the resulting response will be serialized and inlined into the rendered HTML. This allows the subsequent client-side `load` to access identical data immediately without an additional network request.
+
+#### session
+
+`session` can be used to pass data from the server related to the current request, e.g. the current user. By default it is `undefined`. [Seeding session data](#Seeding_session_data) describes how to add data to it.
+
+#### context
+
+`context` is passed from layout components to child layouts and page components. For the root `$layout.svelte` component, it is equal to `{}`, but if that component's `load` function returns an object with a `context` property, it will be available to subsequent `load` functions.
+
+> It is important to note that `load` may run on either the server or in the client browser. Code called inside `load` blocks:
```

Comment:
they're run once per request. I wasn't aware of that behaviour in Sapper — I think we'd need to think about that API a bit more. It probably warrants a separate issue but I'm inclined to file it under YAGNI until someone pops up with a credible use case. I don't think `load` functions should ever have side-effects

Reply:
That behavior sounds more correct to me, as well. I think I was just worried about fixing it in Sapper because what if it's someone's use case? I feel a lot safer about fixing this going from Sapper to Kit. I'm pretty sure people could still just do side-effect-y stuff somewhere in a subscription to the page store if they needed to.